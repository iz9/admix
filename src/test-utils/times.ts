export function times(amount: number, cb: (...args: any) => any) {
  let i = amount
  while (i > 0) {
    !void cb() && --i
  }
}
