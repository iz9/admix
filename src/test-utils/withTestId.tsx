import React from 'react'

interface WithTestID {
  testId: string
}

export function withTestId<P>(Component: React.ComponentType<P>) {
  const WithTestId: React.FC<P & WithTestID> = props => {
    const { testId, ...rest } = props
    return <Component data-testid={testId} {...(rest as P)} />
  }
  return WithTestId
}
