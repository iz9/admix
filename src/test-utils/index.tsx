import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { withTestId } from './withTestId'
import { ThemeProvider } from '../ui-lib/theme'
import { hexToRGB } from './hexToRgb'

const AllTheProviders: React.FC = ({ children }) => (
  <ThemeProvider>{children as React.ReactChild}</ThemeProvider>
)

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render, withTestId, hexToRGB }
