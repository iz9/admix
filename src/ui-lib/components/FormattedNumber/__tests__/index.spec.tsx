import React from 'react'
import { cleanup, render, withTestId, screen } from 'test-utils'

import { FormattedNumber } from '..'

describe('<FormattedNumber  />', () => {
  const data = {
    numberAsIs: { source: 29283774, expected: '29,283,774' },
    compactLT1K: { source: 788, expected: '788' },
    compactMT1K: { source: 78928, expected: '78.9K' },
    compactMT1M: { source: 78928928, expected: '78.9M' },
    compactMT1B: { source: 78928928273, expected: '78.9B' },
    currency: { source: 8.24, expected: '$8.24' },
    percent: { source: 69, expected: '69%' },
  }
  afterEach(() => {
    cleanup()
  })

  it('should match snapshot', () => {
    const { container } = render(<FormattedNumber value={12345} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render numbers formatted as it is', () => {
    const { container } = render(
      <FormattedNumber value={data.numberAsIs.source} />,
    )
    expect(container.firstChild?.textContent).toBe(data.numberAsIs.expected)
  })
  it('should render number in compact mode', () => {
    render(
      <div>
        <div data-testid={data.compactLT1K.expected}>
          <FormattedNumber value={data.compactLT1K.source} variant="compact" />
        </div>
        <div data-testid={data.compactMT1K.expected}>
          <FormattedNumber value={data.compactMT1K.source} variant="compact" />
        </div>
        <div data-testid={data.compactMT1M.expected}>
          <FormattedNumber value={data.compactMT1M.source} variant="compact" />
        </div>
        <div data-testid={data.compactMT1B.expected}>
          <FormattedNumber value={data.compactMT1B.source} variant="compact" />
        </div>
      </div>,
    )

    expect(screen.getByTestId(data.compactLT1K.expected).innerHTML).toBe(
      data.compactLT1K.expected,
    )
    expect(screen.getByTestId(data.compactMT1K.expected).innerHTML).toBe(
      data.compactMT1K.expected,
    )
    expect(screen.getByTestId(data.compactMT1M.expected).innerHTML).toBe(
      data.compactMT1M.expected,
    )
    expect(screen.getByTestId(data.compactMT1B.expected).innerHTML).toBe(
      data.compactMT1B.expected,
    )
  })

  it('should render number as percents', () => {
    render(
      <div data-testid={data.percent.expected}>
        <FormattedNumber value={data.percent.source} variant="percent" />
      </div>,
    )

    expect(screen.getByTestId(data.percent.expected).innerHTML).toBe(
      data.percent.expected,
    )
  })

  it('should render number as currency', () => {
    render(
      <div data-testid={data.currency.expected}>
        <FormattedNumber value={data.currency.source} variant="currency" />
      </div>,
    )

    expect(screen.getByTestId(data.currency.expected).innerHTML).toBe(
      data.currency.expected,
    )
  })
})
