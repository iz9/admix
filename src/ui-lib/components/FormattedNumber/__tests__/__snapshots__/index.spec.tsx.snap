// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedNumber  /> should match snapshot 1`] = `12,345`;
